packages/shared/src/UI/contexts/PageUIProvider.tsx

```diff
@@ -25,11 +26,25 @@ interface MaskUIRootProps extends React.PropsWithChildren<{}> {
     fallback?: React.ReactNode
 }
 
+const persister = createSyncStoragePersister({
+    storage: window.localStorage,
+})
+// We don't persist all queries but only those have the first key starts with '@@'
+const persistOptions: Omit<PersistQueryClientOptions, 'queryClient'> = {
+    persister,
+    dehydrateOptions: {
+        shouldDehydrateQuery: ({ queryKey }) => {
+            if (typeof queryKey[0] !== 'string') return false
+            return queryKey[0].startsWith('@@')
+        },
+    },
+}
+
 function MaskUIRoot({ children, useTheme, fallback }: MaskUIRootProps) {
     return (
         <DialogStackingProvider hasGlobalBackdrop={false}>
             <MaskThemeProvider useMaskIconPalette={(theme) => theme.palette.mode} useTheme={useTheme}>
-                <QueryClientProvider client={queryClient}>
+                <PersistQueryClientProvider client={queryClient} persistOptions={persistOptions}>
```

Comment:
I don't think we can add it here. It will store all data in, for example, twitter.com. Other plugins/twitter can access/modify them as well. Please only add it in `PageUIProvider`, not in `SiteUIProvider`.

Reply:
This **is** `PageUIProvider`.

Reply:
No, this is `MaskUIRoot` which is shared by both providers.

<img width="356" alt="image" src="https://github.com/DimensionDev/Maskbook/assets/5390719/ef0a4237-d9de-4534-8172-322d6be11e77">

Reply:
Not really, they have their own `<MaskUIRoot>`

Reply:
oh sorry I didn't notice that